core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -72,6 +79,7 @@ class Replica(val brokerId: Int,
     else if (logReadResult.info.fetchOffsetMetadata.messageOffset >= lastFetchLeaderLogEndOffset)
       _lastCaughtUpTimeMs = math.max(_lastCaughtUpTimeMs, lastFetchTimeMs)
 
+    logStartOffset = logReadResult.leaderLogStartOffset
```

Comment:
Oops... Good point! Thanks! I fixed the bug by passing `followerLogStartOffset` to `updateLogReadResult` via `LogReadResult`.